### DISCUSSION TITLE
UE not registered to srsran gnb

### DISCUSSION THREAD

**Discussion by hayyanali-cmd** [srsran/srsRAN_Project#876] [General Help]
Hello srsran_project team

This is first time for me with srsran_project, i worked before alot on srsran_4G
I installed srsran_project from source last week, followed the instructions in tutorials, I am using pixel 6a (SIM plmnid is 00101) and usrp b210. as you can see:
*************
sudo gnb -c gnb_rf_b200_tdd_n78_20mhz.yml 

--== srsRAN gNB (commit e73b46182) ==--

The PRACH detector will not meet the performance requirements with the configuration {Format B4, ZCZ 0, SCS 30kHz, Rx ports 1}.
Lower PHY in quad executor mode.
Available radio types: uhd.
[INFO] [UHD] linux; GNU C++ version 11.4.0; Boost_107400; UHD_4.7.0.0-0ubuntu1~jammy1
[INFO] [LOGGING] Fastpath logging disabled at runtime.
Making USRP object with args 'type=b200,num_recv_frames=64,num_send_frames=64'
[INFO] [B200] Detected Device: B210
[INFO] [B200] Operating over USB 3.
[INFO] [B200] Detecting internal GPSDO.... 
[INFO] [GPS] Found an internal GPSDO: GPSTCXO, Firmware Rev 0.929b
[INFO] [B200] Initialize CODEC control...
[INFO] [B200] Initialize Radio control...
[INFO] [B200] Performing register loopback test... 
[INFO] [B200] Register loopback test passed
[INFO] [B200] Performing register loopback test... 
[INFO] [B200] Register loopback test passed
[INFO] [B200] Setting master clock rate selection to 'automatic'.
[INFO] [B200] Asking for clock rate 16.000000 MHz... 
[INFO] [B200] Actually got clock rate 16.000000 MHz.
[INFO] [MULTI_USRP] Setting master clock rate selection to 'manual'.
[INFO] [B200] Asking for clock rate 23.040000 MHz... 
[INFO] [B200] Actually got clock rate 23.040000 MHz.
Cell pci=1, bw=20 MHz, 1T1R, dl_arfcn=627340 (n78), dl_freq=3410.1 MHz, dl_ssb_arfcn=626976, ul_freq=3410.1 MHz

N2: Connection to AMF on 127.0.1.100:38412 completed
==== gNB started ===
Type <h> to view help
Late: 0; Underflow: 1; Overflow: 0;
Late: 0; Underflow: 1; Overflow: 0;
****************************

here is the AMF config file:
![Screenshot from 2024-10-17 11-59-35](https://github.com/user-attachments/assets/4c099553-acf4-4a10-ac26-15adbff0c12e)
and here is the UPF config:
![Screenshot from 2024-10-17 12-01-31](https://github.com/user-attachments/assets/21f69a18-b005-43c6-8c6e-8aef5ceffa53)
here is gnb file config:
![Screenshot from 2024-10-17 12-05-21](https://github.com/user-attachments/assets/adeaf05f-2bda-4b35-a843-cd10085e6800)

# Expected behavior: 
the UE is attached properly 

# Actual behavior: 
I can see my network but cant connect, when checking the gnb_mac.pcap using wireshark: the registration reject cause is :  "5GMM cause: Semantically incorrect message (95)". 
![Screenshot from 2024-10-17 12-07-52](https://github.com/user-attachments/assets/81e05f0e-1c0c-4bf8-9477-91423368c101)

here is the amf log:
 ***********
 10:18:09.384: [amf] INFO: [Added] Number of gNB-UEs is now 1 (../src/amf/context.c:2694)
10/17 10:18:09.384: [amf] INFO:     RAN_UE_NGAP_ID[0] AMF_UE_NGAP_ID[1] TAC[7] CellID[0x66c000] (../src/amf/ngap-handler.c:596)
10/17 10:18:09.384: [amf] INFO: [suci-0-001-01-0-0-0-0123456789] Unknown UE by SUCI (../src/amf/context.c:1861)
10/17 10:18:09.384: [amf] INFO: [Added] Number of AMF-UEs is now 1 (../src/amf/context.c:1638)
10/17 10:18:09.384: [gmm] INFO: Registration request (../src/amf/gmm-sm.c:1220)
10/17 10:18:09.384: [gmm] INFO: [suci-0-001-01-0-0-0-0123456789]    SUCI (../src/amf/gmm-handler.c:174)
10/17 10:18:09.386: [gmm] ERROR: [suci-0-001-01-0-0-0-0123456789] HTTP response error [400] (../src/amf/gmm-sm.c:1800)
10/17 10:18:09.386: [amf] WARNING: [suci-0-001-01-0-0-0-0123456789] Registration reject [95] (../src/amf/nas-path.c:211)
10/17 10:18:09.523: [amf] INFO: UE Context Release [Action:3] (../src/amf/ngap-handler.c:1726)
10/17 10:18:09.523: [amf] INFO:     RAN_UE_NGAP_ID[0] AMF_UE_NGAP_ID[1] (../src/amf/ngap-handler.c:1727)
10/17 10:18:09.523: [amf] INFO:     SUCI[suci-0-001-01-0-0-0-0123456789] (../src/amf/ngap-handler.c:1731)
10/17 10:18:09.523: [amf] INFO: [Removed] Number of gNB-UEs is now 0 (../src/amf/context.c:2701)
10/17 10:18:09.523: [amf] INFO: [Removed] Number of AMF-UEs is now 0 (../src/amf/context.c:1735)
10/17 10:19:26.651: [amf] INFO: gNB-N2[127.0.0.1] connection refused!!! (../src/amf/amf-sm.c:855)
10/17 10:19:26.651: [amf] INFO: [Removed] Number of gNBs is now 0 (../src/amf/context.c:1265)

**************************************

It is strange, because i am pretty sure that configs of the UE SIM (programmable osmocom one) are the same of configs of UE in open5gs webui. 
![Screenshot from 2024-10-17 10-58-18](https://github.com/user-attachments/assets/03b14a96-eb6d-4704-a274-030d3e498f8b)
![Screenshot from 2024-10-17 10-58-41](https://github.com/user-attachments/assets/2d2d55b3-37a0-46c3-a736-e75a22b7d744)

I ask if you can suggest thankfully a solution to this problem :), all files are attached here:
[registration_problem.zip](https://github.com/user-attachments/files/17410447/registration_problem.zip)

**Comment by newt2105** [comment]
Hi [hayyanali-cmd](https://github.com/hayyanali-cmd),
Did you solve this problem?

**Reply by hayyanali-cmd** [reply]
yes, it is , just by disabling all 4G services like MME, .. also it worked only on my Samsung A25 device
